ExpireEycaActivity/handler.ts

```diff
@@ -21,49 +23,65 @@ export const getExpireEycaActivityHandler = (
   logPrefix: string = "ExpireEycaActivity"
 ) => (context: Context, input: unknown): Promise<ActivityResult> => {
   const fail = failure(context, logPrefix);
-  return fromEither(ActivityInput.decode(input))
-    .mapLeft(errs => fail(errorsToError(errs), "Cannot decode Activity Input"))
-    .chain(activityInput =>
-      userEycaCardModel
-        .findLastVersionByModelId([activityInput.fiscalCode])
-        .mapLeft(() =>
+  return pipe(
+    input,
+    ActivityInput.decode,
+    TE.fromEither,
+    TE.mapLeft(errs =>
+      fail(errorsToError(errs), "Cannot decode Activity Input")
+    ),
+    TE.chain(activityInput =>
+      pipe(
+        userEycaCardModel.findLastVersionByModelId([activityInput.fiscalCode]),
+        TE.mapLeft(() =>
           fail(
             new Error(
               "Cannot retrieve User EYCA Card for the provided fiscalCode"
             )
           )
-        )
-        .chain(maybeUserEycaCard =>
-          fromEither(
-            fromOption(
+        ),
+        TE.chain(maybeUserEycaCard =>
+          pipe(
+            maybeUserEycaCard,
+            E.fromOption(() =>
               fail(
                 new Error("No User EYCA Card found for the provided fiscalCode")
               )
-            )(maybeUserEycaCard)
+            ),
+            TE.fromEither
```

Comment:
In case you accept previous suggestion

```suggestion
            )
```